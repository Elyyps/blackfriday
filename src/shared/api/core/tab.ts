export class Tab {
  public title: string[];
}
