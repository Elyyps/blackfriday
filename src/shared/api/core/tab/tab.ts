export class Tab {
  public content: any;
  public index: number;
  public title: string;
}
