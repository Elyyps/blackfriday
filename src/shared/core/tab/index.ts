export * from './tab-component.component'