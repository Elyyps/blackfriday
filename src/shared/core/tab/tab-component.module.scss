.tab {
  div {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
  }
  a {
    width: 112px;
    height: 49px;
    font-size: 16px;
    display: flex;
    align-items: center;
    justify-content: center;
    @media (max-width: $breakpoint-small-medium) {
      width: 50%;
    }
  }

  &__selected {
    background-color: rgba($brand-primary, 0.1);
    border-bottom: 2px solid $brand-primary !important;
    color: $brand-primary !important;
  }
  .uk-tab {
    margin: 0;
    padding: 0;
    list-style: none;
    &__list {
      display: flex;
      align-items: center;
      justify-content: center;
      margin: 0;
      > a {
        font-weight: 600;
        a {
          padding: 14px 16px;
          text-decoration: none;
          color: inherit;
          display: inline-flex;
        }
        &.uk-active {
          a {
            background-color: rgba($brand-primary, 0.1);
            position: relative;
            color: $brand-primary;
            &:after {
              content: "";
              width: 100%;
              bottom: 0;
              position: absolute;
              background-color: $brand-primary;
              height: 2px;
              left: 0;
            }
          }
        }
      }
    }
  }
}
