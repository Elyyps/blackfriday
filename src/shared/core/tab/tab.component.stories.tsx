// import * as React from "react";

// import { storiesOf } from "@storybook/react";
// import { withA11y } from "@storybook/addon-a11y";
// import { withProvider } from "@app/util";
// import { TabComponent, ITabsInterface } from "./tab.component";
// import { StoresOverviewContainer } from "@app/modules/stores-overview/container/stores-overview.container";
// const data: ITabsInterface[] = [
//   { title: "Winkels", content: <StoresOverviewContainer />, index: 1 },
//   { title: "Productdeals", content: <h1>Will be implemented</h1>, index: 2 }
// ];
// storiesOf("tabComponent", module)
//   .addDecorator(withA11y)
//   .addDecorator(withProvider)
//   .add("Basic implementation", () => <TabComponent tabs={data} />);
