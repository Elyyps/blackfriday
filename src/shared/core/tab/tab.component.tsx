import * as React from "react";
import styles from "./tab-component.module.scss";
import { Link } from "react-router-dom";
export interface ITabsInterface {
  content: JSX.Element;
  index: number;
  title: string;
}
export interface ITabComponentProps {
  tabs: any[];
}
const TabComponent = (props: ITabComponentProps) => {
  const [selectedTab, setSelectedTab] = React.useState<number>(1);
  const getComponentTabIndex = () => {
    let tabComponent;
    props.tabs.map(tab => {
      if (selectedTab === tab.index) {
        tabComponent = tab.content;
      }
    });

    return tabComponent;
  };

  return (
    <React.Fragment>
      <div className={`${styles["tab"]}`}>
        <ul className={styles["uk-tab__list"]}>
          {props.tabs.map(tab => (
            <li
              role="button"
              className={selectedTab === tab.index ? styles["tab__selected"] : ""}
              onClick={() => setSelectedTab(tab.index)}
            >
              {<Link to="#">{tab.title}</Link>}
            </li>
          ))}
        </ul>
      </div>
      {getComponentTabIndex()}
    </React.Fragment>
  );
};
export { TabComponent };
