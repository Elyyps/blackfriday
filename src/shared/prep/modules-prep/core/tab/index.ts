export * from "./tab.component";
export * from "./tab-container.component";
