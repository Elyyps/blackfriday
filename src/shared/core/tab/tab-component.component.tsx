/* tslint:disable */
import * as React from "react";
import styles from "./tab-component-component.module.scss";
import { Link } from "react-router-dom";
import { generateTabsDummyData } from "@app/api/core/tab/endpoint";
import { StoresOverviewContainer } from "@app/modules/stores-overview/container/stores-overview.container";

export interface ITabComponentProps {}
const TabComponent = (props: ITabComponentProps) => {
  const tabData = generateTabsDummyData;
  const [selectedTab, setSelectedTab] = React.useState<string>("Winkels");
  const getComponentTabIndex = (title: string) => {
    let tabComponent;

    switch (title) {
      case "Winkels":
        tabComponent = <StoresOverviewContainer />;
    }

    return tabComponent;
  };

  return (
    <React.Fragment>
      <div className={`${styles["tab"]}`}>
        <ul className={styles["uk-tab__list"]}>
          {tabData.title.map((title: string) => (
            <li className={selectedTab === title ? styles["tab__selected"] : ""} onClick={() => setSelectedTab(title)}>
              {<Link to="#">{title}</Link>}
            </li>
          ))}
        </ul>
      </div>
      {getComponentTabIndex(selectedTab)}
    </React.Fragment>
  );
};
export { TabComponent };
