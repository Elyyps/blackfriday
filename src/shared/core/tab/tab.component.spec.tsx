import React from "react";
import { shallow } from "enzyme";
import { TabComponent } from "./tab.component";
const titles: string[] = ["Winkels", "Productdeals"];

describe("[tabComponent]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<TabComponent titles={titles} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
