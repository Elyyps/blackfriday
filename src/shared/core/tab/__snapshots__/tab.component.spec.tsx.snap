// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[tabComponent] should render component without crashing 1`] = `
<Fragment>
  <div
    className="tab"
  >
    <ul
      className="uk-tab__list"
    >
      <li
        className=""
        onClick={[Function]}
        role="button"
      >
        <Link
          to="#"
        >
          Winkels
        </Link>
      </li>
      <li
        className=""
        onClick={[Function]}
        role="button"
      >
        <Link
          to="#"
        >
          Productdeals
        </Link>
      </li>
    </ul>
  </div>
</Fragment>
`;
